test/SocketTest.php

```diff
@@ -28,4 +28,16 @@ public function testAsk() {
             $this->assertSame(MessageTypes::RESPONSE, $result->getType());
         });
     }
+
+    public function testGetLastActivity() {
+        Loop::run(function () {
+            $question = (new QuestionFactory)->create(Dns\Record::A);
+            $question->setName("google.com");
+
+            /** @var Dns\Internal\Socket $socket */
+            $socket = yield $this->connect();
+
+            $this->assertRegExp("/152061[0-9][0-9][0-9][0-9]/", (string) $socket->getLastActivity());
```

Comment:
Why that assertion? Should be enough to assert `$socket->getLastActivity() < time() + 1`.